test/OpenTelemetry.Contrib.Instrumentation.GrpcCore.Tests/GrpcCoreClientInterceptorTests.cs

```diff
@@ -0,0 +1,463 @@
+
+            // Cancelled is not an error.
+            if (expectedStatusCode != StatusCode.OK && expectedStatusCode != StatusCode.Cancelled)
+            {
+                Assert.IsTrue(activity.TagObjects.Any(
+                    t => t.Key == SemanticConventions.AttributeOtelStatusCode && (string)t.Value == "ERROR"));
+            }
+
+            if (recordedMessages)
+            {
+                // all methods accept a request and return a single response
+                Assert.IsNotNull(activity.Events);
+                var requestMessage = activity.Events.FirstOrDefault(ae => ae.Name == FoobarService.DefaultRequestMessage.GetType().Name);
+                var responseMessage = activity.Events.FirstOrDefault(ae => ae.Name == FoobarService.DefaultResponseMessage.GetType().Name);
+
+                static void ValidateCommonEventAttributes(ActivityEvent activityEvent)
+                {
+                    Assert.IsNotNull(activityEvent.Tags);
+                    Assert.IsTrue(activityEvent.Tags.Any(t => t.Key == "name" && (string)t.Value == "message"));
+                    Assert.IsTrue(activityEvent.Tags.Any(t => t.Key == SemanticConventions.AttributeMessageID && (int)t.Value == 1));
+                }
+
+                Assert.IsNotNull(requestMessage);
+                Assert.IsNotNull(responseMessage);
+
+                ValidateCommonEventAttributes(requestMessage);
+                Assert.IsTrue(requestMessage.Tags.Any(t => t.Key == SemanticConventions.AttributeMessageType && (string)t.Value == "SENT"));
+                Assert.IsTrue(requestMessage.Tags.Any(t => t.Key == SemanticConventions.AttributeMessageCompressedSize && (int)t.Value == FoobarService.DefaultRequestMessageSize));
+
+                ValidateCommonEventAttributes(responseMessage);
+                Assert.IsTrue(responseMessage.Tags.Any(t => t.Key == SemanticConventions.AttributeMessageType && (string)t.Value == "RECEIVED"));
+                Assert.IsTrue(requestMessage.Tags.Any(t => t.Key == SemanticConventions.AttributeMessageCompressedSize && (int)t.Value == FoobarService.DefaultResponseMessageSize));
+            }
+        }
+
+        /// <summary>
+        /// Tests basic handler success.
+        /// Also validates additional Activity tags written via the OnRpcStarted callback.
+        /// </summary>
+        /// <param name="clientRequestFunc">The client request function.</param>
+        /// <returns>A Task.</returns>
+        private async Task TestHandlerSuccess(Func<Foobar.FoobarClient, Task> clientRequestFunc)
+        {
+            var interceptorOptions = new ClientTracingInterceptorOptions
+            {
+                Propagator = new TraceContextPropagator(),
+                RecordMessageEvents = true,
+            };
+
+            // No Activity parent
+            using (var activityListener = new InterceptorActivityListener())

```

Comment:
it'd be good to have at least one integrated test which uses actual OpenTelemetrySdk, than own Listener to validate the created activity.

Reply:
Will do

Reply:
I added an E2E example that utilizes the OpenTelemetry SDK.